File: spring-web-java/src/test/java/spring/web/java/infrastructure/config/DataJpaTestConfig.java
package spring.web.java.infrastructure.config;

import org.mockito.Mockito;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;

import com.querydsl.jpa.impl.JPAQueryFactory;

import jakarta.persistence.EntityManager;
import spring.web.java.infrastructure.util.crypto.CryptoUtil;

@TestConfiguration
public class DataJpaTestConfig {

	@Bean
	public JPAQueryFactory jpaQueryFactory(EntityManager entityManager) {
		return new JPAQueryFactory(entityManager);
	}

	@Bean
	public CryptoUtil cryptoUtil() {
		return Mockito.mock(CryptoUtil.class);
	}
}

File: spring-web-kotlin/src/test/kotlin/spring/web/kotlin/infrastructure/config/DataJpaTestConfig.kt
package spring.web.kotlin.infrastructure.config

import com.querydsl.jpa.impl.JPAQueryFactory
import io.mockk.mockk
import jakarta.persistence.EntityManager
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean
import spring.web.kotlin.infrastructure.util.crypto.CryptoUtil

@TestConfiguration
class DataJpaTestConfig {

    @Bean
    fun jpaQueryFactory(entityManager: EntityManager): JPAQueryFactory = JPAQueryFactory(entityManager)

    @Bean
    fun cryptoUtil() = mockk<CryptoUtil>()
}